File: src/com/sorting/selection_sort.java
package com.sorting;

import java.util.Arrays;

public class selection_sort {
    public static void main(String[] args) {
        int[] arr = {6,0,5,2,3,1};
        int N = arr.length;
        for (int i = 0; i <= N-1; i++) {
            int res = getMaxInt(arr, 0, N-1-i);
            swapNum(arr, N-1-i, res);
        }
        System.out.println(Arrays.toString(arr));
    }

    public static int getMaxInt(int[] arr, int start, int end){
        int res = start;
        for (int i = start; i <= end; i++) {
            if (arr[res] < arr[i]) res = i;
        }
        return res;
    }

    public static void swapNum(int[] arr, int target, int index){
        int temp = arr[index];
        arr[index] = arr[target];
        arr[target] = temp;
    }
}

File: js_dsa/Recursion/selection_sort.js
const arr = [5, 2, 3, 1, 4, 6];

function sort(arr, r, c, max) {
    if (r === 0) {
        return;
    }

    if (c < r) {
        if (max > -1) {
            if (arr[c] > arr[max]) {
                max = c;
            }
        } else {
            max = c;
        }
        return sort(arr, r, c + 1, max);
    } else {
        swap(arr, r, max);
        return sort(arr, r - 1, 0, -1);
    }
}

function swap(arr, from, to) {
    let temp = arr[from];
    arr[from] = arr[to];
    arr[to] = temp;
}

sort(arr, arr.length - 1, 0, -1);
console.log(arr);
